pilot/pkg/model/push_context.go

```diff
@@ -2217,14 +2217,19 @@ func (ps *PushContext) ReferenceAllowed(kind config.GroupVersionKind, resourceNa
 	// Currently, only Secret has reference policy, and only implemented by Gateway API controller.
 	switch kind {
 	case gvk.Secret:
-		if ps.GatewayAPIController != nil {
-			return ps.GatewayAPIController.SecretAllowed(resourceName, namespace)
-		}
+		ps.secretAllowed(resourceName, namespace)
 	default:
 	}
 	return false
 }
 
+func (ps *PushContext) secretAllowed(resourceName string, namespace string) bool {
+	if ps.GatewayAPIController != nil {
+		return ps.GatewayAPIController.SecretAllowed(resourceName, namespace)
```

Comment:
Using SecretAllowed here doesn't seem appropriate. The code is checking for a reference `From KubernetesGateway -> To Secret`. But this is WASM -- its not referenced from a KubernetesGateway but a WASMPlugin

Reply:
I think - double checking with gateway folks

Reply:
You are right. I did not look at the implementation, It is hard coding kubernetes gateway.,

Reply:
Also just realized we do not process any other reference grants other than gateway resources [here](https://github.com/istio/istio/blob/5a1314991c1daef5249888015fff0e0c46eda03e/pilot/pkg/config/kube/gateway/conversion.go#L184). So if we go down this path this needs to be enhanced as well.